sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1139,36 +1140,33 @@ class Analyzer(override val catalogManager: CatalogManager)
      */
     private def lookupTableOrView(
         identifier: Seq[String],
-        viewOnly: Boolean = false): Option[LogicalPlan] = {
+        viewOnly: Boolean = false): LogicalPlan = {
```

Comment:
Hmm, so if `lookupTableOrView` cannot resolve the table/view, it directly throws an exception now?

Reply:
Yes, I think we over-designed this in the first place. We can't fail immediately for `UnresolvedRelation` because there is a rule `ResolveSQLOnFile` that can try to resolve it one more time. We don't have such a requirement for DDL commands.